crates/rover-client/src/operations/subgraph/check_workflow/runner.rs

```diff
@@ -78,30 +80,40 @@ fn get_check_response_from_data(
 
     let workflow_status = check_workflow.status;
     let mut operations_status = None;
-    let mut operations_target_url = None;
     let mut operations_result = None;
     let mut number_of_checked_operations: u64 = 0;
     let mut core_schema_modified = false;
     let mut composition_errors = Vec::new();
     let mut downstream_status = None;
     let mut downstream_target_url = None;
     let mut blocking_downstream_variants = Vec::new();
+
+    // This will either be the operation check target url if present or the composition check target url if present or None if none are found
+    let mut display_target_url = None;
     for task in check_workflow.tasks {
         match task.on {
             OperationsCheckTask(typed_task) => {
                 operations_status = Some(task.status);
-                operations_target_url = task.target_url;
+                display_target_url = task.target_url;
                 if let Some(result) = typed_task.result {
                     number_of_checked_operations =
                         result.number_of_checked_operations.try_into().unwrap();
                     operations_result = Some(result);
+                } else {
+                    // We can early exit because we know this is a race condition and throw an error
+                    return Err(RoverClientError::AdhocError {
+                        msg: "Operations check task has no result.".to_string(),
+                    });
                 }
```

Comment:
this is great and the `match task.on` gives me high confidence that we won't give folks a passing check for a check that didn't pass👍🏼